### PR TITLE
upgrade tools, add fix and test for #90

### PR DESCRIPTION
**Pull request by SanderElias**
In this PR I upgrade TS to 4.7.2,
moved  build to use wireit to aid building and testing.
Added a test to test for the issue form #90 
and fixed the above test

### CODE REVIEW

**Comment by kenchris** [comment]
what is up with the above failing and cancelled tests?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "urlpattern-polyfill",
-  "version": "5.0.0",
+  "version": "5.0.0-dev.4a67043",
```
Comment (kenchris) [inline package.json]: remove?

Comment (SanderElias) [reply package.json]: updated

File: package.json [hunk 2]
```diff
@@ -53,15 +54,76 @@
...
-    "build": "npm run build:esm && npm run build:cjs && npm run copy:dts",
-    "pretest": "npm run build; rm -rf node_modules/urlpattern-polyfill; ln -s $(pwd) node_modules/urlpattern-polyfill",
-    "test": "ava --fail-fast -s",
-    "manual-test": "npx http-server -o /index.html -p 4203",
-    "publish-dev": "npm test && VERSION=${npm_package_version%-*}-dev.`git rev-parse --short HEAD` && npm version --no-git-tag-version $VERSION && npm publish --tag dev"
+    "build:esm": "wireit",
+    "build:cjs": "wireit",
+    "build": "wireit",
+    "copyTypeFiles": "wireit",
+    "prepFakeNodeModules": "wireit",
```
Comment (kenchris) [inline package.json:61]: Do you need this here when you are not going to call it manually from the command line?

**Comment by SanderElias** [comment]
@kenchris Tests are using top-level await (I didn't change that in this PR)
But node 12 doesn't, so the test did fail. Removed V12 for testing as its unsupported by node now anyway

File: package.json (hunk 2 above)
Comment (SanderElias) [reply package.json:61]: It seems it's mandatory. I was wondering the same thing.

**Review by kenchris — APPROVED** [review]

**Pull request merged**